src/components/delete-with-confirmation-button.tsx

```diff
@@ -0,0 +1,52 @@
+import React from 'react';
+
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogContentText,
+  DialogActions
+} from '@mui/material';
+
+export const DeleteWithConfirmationButton = (props: {
+  handleDelete: () => Promise<void>;
+  title: string;
+  text?: string;
+}): JSX.Element => {
+  const [open, setOpen] = React.useState(false);
+
+  const handleClose = () => {
+    setOpen(false);
+  };
+
+  return (
+    <>
+      <Button variant="contained" color="error" onClick={_ => setOpen(true)}>
+        {props.title}
+      </Button>
+      <Dialog open={open} onClose={handleClose}>
+        <DialogTitle>{props.title}</DialogTitle>
+        {props.text && (
+          <DialogContent>
+            <DialogContentText>{props.text}</DialogContentText>
+          </DialogContent>
+        )}
+        <DialogActions>
+          <Button
+            variant="outlined"
+            onClick={_ => {
+              handleClose();
+              props.handleDelete();
+            }}
+          >
+            Yes
```

Comment:
JupyterLab tends to use the [Apple Human Interface Guidelines](https://developer.apple.com/design/human-interface-guidelines/components/presentation/alerts/): the primary button is on the right, and "Yes" and "No" are discouraged in favor of action verbs. In this case, the primary button should be "Delete" (localized) and the secondary button should be "Cancel".

Reply:
Updated:
![Screen Shot 2022-10-13 at 4 55 30 PM](https://user-images.githubusercontent.com/26686070/195732682-43dcbe73-7114-4f34-8b74-7383913b13d6.png)

Reply:
The primary button is typically colored in in our dialogs; let's make the delete button red, like in this example dialog. Also note that the selected button is "Cancel," so if the user hits <kbd>SPACE</kbd> or <kbd>ENTER</kbd>, they will take the non-destructive action.

![image](https://user-images.githubusercontent.com/93281816/195732906-b385f57e-49ce-497f-a39e-7e2ccaca4f15.png)

Reply:
Updated versions of the dialog:
![Screen Shot 2022-10-13 at 5 58 47 PM](https://user-images.githubusercontent.com/26686070/195738401-ae1d9556-1239-4305-a7d1-817aa5571530.png)
![Screen Shot 2022-10-13 at 5 58 35 PM](https://user-images.githubusercontent.com/26686070/195738402-b368734f-5ead-4d50-b0b0-e1959ed96496.png)